velox/connectors/hive/benchmarks/HivePartitionFunctionBenchmark.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include "velox/connectors/hive/HivePartitionFunction.h"
+#include "velox/functions/lib/benchmarks/FunctionBenchmarkBase.h"
+#include "velox/type/Type.h"
+#include "velox/vector/ComplexVector.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+namespace facebook::velox::test {
+namespace {
+
+void BM_HivePartFunc(
+    size_t,
+    size_t vectorSize,
+    size_t bucketCount,
+    const TypePtr& type,
+    size_t stringLength = 0) {
+  functions::test::FunctionBenchmarkBase benchmarkBase;
+  folly::BenchmarkSuspender suspender;
+
+  VectorFuzzer::Options opts;
```

Comment:
It would be nice to separate setup logic from the logic being measured. Consider creating a class and put setup logic into constructor and logic being measured into run method.  See velox/benchmarks/basic/SimpleArithmetic.cpp for an example.